lib/ansible/plugins/inventory/docker_machine.py

```diff
@@ -0,0 +1,188 @@
+
+import json
+import re
+import subprocess
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    ''' Host inventory parser for ansible using Docker machine as source. '''
+
+    NAME = 'docker_machine'
+
+    def _run_command(self, *args):
+        return subprocess.check_output(["docker-machine"] + list(args)).decode('utf-8').strip()
+
+    def _set_docker_daemon_variables(self, id):
+        '''
+        Capture settings from Docker Machine that would be needed to connect to the remote Docker daemon installed on
+        the Docker Machine remote host. Note: passing '--shell=sh' is a workaround for 'Error: Unknown shell'.
+        '''
+        env_out = self._run_command('env', '--shell=sh', id)
+
+        # example output of docker-machine env --shell=sh:
+        #   export DOCKER_TLS_VERIFY="1"
+        #   export DOCKER_HOST="tcp://134.209.204.160:2376"
+        #   export DOCKER_CERT_PATH="/root/.docker/machine/machines/routinator"
+        #   export DOCKER_MACHINE_NAME="routinator"
+        #   # Run this command to configure your shell:
+        #   # eval $(docker-machine env --shell=bash routinator)
+
+        # capture any of the DOCKER_xxx variables that were output and create Ansible host vars
+        # with the same name and value but with a dm_ name prefix.
+        for line in env_out.splitlines():
+            match = re.search('(DOCKER_[^=]+)="([^"]+)"', line)
+            if match:
+                env_var_name = match.group(1)
+                env_var_value = match.group(2)
+                self.inventory.set_variable(id, 'dm_{0}'.format(env_var_name), env_var_value)
+
+    def _set_tag_variables(self, id):
+        tags = self.node_attrs['Driver'].get('Tags') or ''
+        self.inventory.set_variable(id, 'dm_tags', tags)
+
+        if tags:
+            split_tags = self.get_option('split_tags')
+            split_separator = self.get_option('split_separator')
+
+            kv_pairs = [kv_pair.strip() for kv_pair in tags.split(',') if kv_pair.strip()]
+            for kv_pair in kv_pairs:
+                if split_tags and split_separator in kv_pair:
+                    k, v = kv_pair.split(split_separator)
+                    self.inventory.set_variable(id, 'dm_tag_{0}'.format(k), v)
+                else:
+                    self.inventory.set_variable(id, 'dm_tag_{0}'.format(kv_pair))
+
+    def _populate(self):
+        self.inventory.add_group('all')
+
+        try:
+            self.nodes = self._run_command('ls', '-q').splitlines()
+            for self.node in self.nodes:
+                self.node_attrs = json.loads(self._run_command('inspect', self.node))
+
+                id = self.node_attrs['Driver']['MachineName']
+                self.inventory.add_host(id)
+                self.inventory.add_group(self.node_attrs['DriverName'])
+                self.inventory.add_host(id, group=self.node_attrs['DriverName'])
+                self.inventory.add_host(id, group='all')
```

Comment:
@felixfontein: Looking at the [pattern docs](https://docs.ansible.com/ansible/latest/user_guide/intro_patterns.html) is it the case that this 'all' group is actually redundant and should be removed from my code?

Reply:
Yes, I think it isn't needed. You can remove it and see if it still works if you limit to `all` :)

Reply:
Removed.